### PR TITLE
Drop ElasticSearch 5 support

### LINKED ISSUE #1658: Drop support for ElasticSearch < 6
ElasticSearch 5 is EOL. We should drop support for it with deprecated mapping multi-types.

### PR DESCRIPTION
**Pull request by porunov**
Fixes #1658

The build is also tested with personal integration tests against ElasticSearch 6.6.0 and ElasticSearch 7.2.0. 

Signed-off-by: Oleksandr Porunov <alexandr.porunov@gmail.com>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [x] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestElasticSearchClient.java [hunk 1]
```diff
@@ -90,7 +90,7 @@
         mapWriter = mapper.writerWithView(Map.class);
     }
 
-    private static final ElasticMajorVersion DEFAULT_VERSION = ElasticMajorVersion.FIVE;
+    private static final ElasticMajorVersion DEFAULT_VERSION = ElasticMajorVersion.SIX;
```
Comment (farodin91) [inline janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestElasticSearchClient.java]: I would directly upgrade to default version 7 or?

Comment (porunov) [reply janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestElasticSearchClient.java]: `DEFAULT_VERSION` is used only when we cannot determine the version from ES which can potentially happen only if we are using unsupported versions or custom ElasticSearch builds. I am OK with any default version, so I've updated it to `SEVEN` as you propose.

Comment (farodin91) [reply janusgraph-es/src/main/java/org/janusgraph/diskstorage/es/rest/RestElasticSearchClient.java]: Thank you for the update.

**Comment by porunov** [comment]
@mbrukman Thank you for this reminder. Updated the table.

**Pull request merged**